modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -529,19 +531,32 @@ else if (Objects.equals(
 
 			throw new ObjectValidationRuleSettingNameException.
 				MissingRequiredName(
-					ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID);
+					ObjectValidationRuleSettingConstants.
+						NAME_OUTPUT_OBJECT_FIELD_ID);
 		}
 
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
+			if (FeatureFlagManagerUtil.isEnabled("LPS-187854") &&
+				!(objectValidationRuleSetting.compareName(
+					ObjectValidationRuleSettingConstants.
+						NAME_KEY_OBJECT_FIELD_ID) ||
+				  objectValidationRuleSetting.compareName(
+					  ObjectValidationRuleSettingConstants.
+						  NAME_OUTPUT_OBJECT_FIELD_ID))) {
```

Comment:
Do we need to also check NAME_OUTPUT_OBJECT_FIELD_ID here? It seems that it's already covered in the if condition below. We should keep NAME_OUTPUT_OBJECT_FIELD_ID working as before, even without this feature flag.